sharktank/tests/models/llama/kv_cache_test.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import unittest
+import torch
+import torch.nn as nn
+from sharktank.models.llama.llama import (
+    PagedLlamaAttentionBlock,
+    PagedKVCache,
+    DirectKVCache,
+)
+from sharktank.models.llama.testing import *
+from sharktank.layers.rotary_embedding import RotaryEmbeddingLayer
+from sharktank.layers import causal_llm
+
+
+class KVCacheTest(unittest.TestCase):
+    def setUp(self):
+        self.block_count = 5
+        self.seq_len = 16
+        self.head_count = 32
+        self.head_dim = 128
+        self.ffn_dim = 11008
+        self.head_count_kv = 32
+        self.block_seq_stride = 16
+        self.rms_epsilon = 1e-5
+        self.rope_dimension_count = 128
+        self.max_seq_len = 4096
+        self.start_positions = torch.tensor([8])
+        self.bs = 1
+        self.attention_block_theta = make_attention_block_theta(
+            feature_dim=self.head_count * self.head_dim,
+            ffn_dim=self.ffn_dim,
+            dtype=torch.float32,
+        )
+        self.paged_kv_cache = PagedKVCache(
+            transformer_block_count=self.head_count,
+            attn_head_count=self.head_count,
+            attn_head_dim=self.head_dim,
+            cache_partition_count=2,  # One for each of K/V.
+            block_seq_stride=self.block_seq_stride,
+            device="cpu",
```

Comment:
probably add self.device instead of using the  raw str

Reply:
Done